extension.js

```diff
@@ -1,47 +1,60 @@
 const cuMarkdown = require('markdown-it-church-slavonic');
-const vscode = require('vscode');
-var markdownit = require('markdown-it');
-
-require('supports-color');
-
-const fs = require("fs");
-
-/**
- * @param {vscode.ExtensionContext} context
- */
-function activate(context) {
-
-    let disposable = vscode.commands.registerCommand('csl.generateXml', function () {  
-        let currentFile =  vscode.window.activeTextEditor.document.uri.fsPath;
-        var xmlRendrer = markdownit().use(cuMarkdown, { renderer: 'xml' });
-        fs.readFile(currentFile, 'utf8', function(err, data) {
-            if (err) throw err;
-            let xml = xmlRendrer.render(data);
-            let newFileName = currentFile.substr(0, currentFile.lastIndexOf('.')) + '.xml';
-            fs.writeFile(newFileName, xml, err => {
-                if (err) {
-                    return vscode.window.showErrorMessage(
-                      "Failed to create boilerplate file!"
-                    );
-                } else {
-                    vscode.window.showInformationMessage('File sucessfully created');
-                    vscode.workspace.openTextDocument(newFileName).then(doc => {
-                        vscode.window.showTextDocument(doc);
-                     });
-                }
-            });
-
-          });
+const { renderXML, renderLaTeX } = cuMarkdown;
+const { window, Uri, workspace, WorkspaceEdit, Position, commands } = require('vscode');
+
+let markdownEngine;  // will be set when Markdown plugin activates
+
+const churchSlavonicGenerate = (render, suffix) => () => {
+    if (window.activeTextEditor === undefined) {
+        return window.showErrorMessage('Please open a Markdown document first');
+    }
+
+    if (window.activeTextEditor.document.languageId !== 'markdown') {
+        return window.showErrorMessage(
+            'This command requires current document to be "markdown", got ' +
+            window.activeTextEditor.document.languageId
+        );
+    }
+
+    if (markdownEngine === undefined) {
+        return window.showErrorMessage(
+            'Internal error - failed to initialize Markdown extension');
```

Comment:
I believe with your code, it will be undefined until a user opens a preview for a document